backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -1158,24 +1168,131 @@ private String getStringFromLocationList(List<String> locationList) {
             .append(" ");
 
     StringBuilder joinClause = new StringBuilder();
+    String repoAlias = "repo";
+    joinClause.append(" INNER JOIN ").append(alias).append(".repository ").append(repoAlias);
     joinClause
         .append(" INNER JOIN ")
         .append(InternalFolderElementEntity.class.getSimpleName())
         .append(" folderElm ")
         .append(" ON ");
     joinClause.append("folderElm.folder_hash = ").append(alias).append(".rootSha ");
 
+    List<KeyValueQuery> predicates = request.getPredicatesList();
     List<String> whereClauseList = new ArrayList<>();
-    if (!request.getRepoIdsList().isEmpty()) {
-      whereClauseList.add(alias + ".repository.id IN (:repoIds) ");
-      parametersMap.put("repoIds", request.getRepoIdsList());
+    if (predicates != null && !predicates.isEmpty()) {
+      for (int index = 0; index < predicates.size(); index++) {
+        KeyValueQuery predicate = predicates.get(index);
+        String[] names = predicate.getKey().split("\\.");
+        switch (names[0]) {
+          case ModelDBConstants.COMMIT:
+            LOGGER.debug("switch case : commit");
+            if (names[1].equals("author")) {
+              whereClauseList.add(alias + "." + names[1] + " = :author");
+              parametersMap.put("author", predicate.getValue().getStringValue());
+            }
+            break;
+          case ModelDBConstants.VERSIONING_REPOSITORY:
+            LOGGER.debug("switch case : " + ModelDBConstants.VERSIONING_REPOSITORY);
+            if (names[1].contains(ModelDBConstants.LABEL)) {
+              joinClause
+                  .append(" INNER JOIN ")
+                  .append(LabelsMappingEntity.class.getSimpleName())
+                  .append(" lb ON lb.id.entity_hash = CAST(")
+                  .append(repoAlias)
+                  .append(".id as string) ")
+                  .append(" AND lb.id.entity_type = :entityType")
+                  .append(" AND lb.id.label = :label");
+              parametersMap.put("entityType", IDTypeEnum.IDType.VERSIONING_REPOSITORY.getNumber());
+              parametersMap.put("label", predicate.getValue().getStringValue());
+            }
+            break;
+          case ModelDBConstants.VERSIONING_COMMIT:
```

Comment:
if i want to filter by commit author predicate key should be COMMIT.author but to filter on commit label it should be VERSIONING_COMMIT.label ?

Reply:
Added java doc comment on the function